src/components/Tasks/Toolbar.tsx

```diff
@@ -23,16 +29,24 @@ interface PropsType {
 export default function Toolbar(props: PropsType) {
   const { defaultCollection, onItemSave, showCompleted, setShowCompleted } = props;
 
-  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const [sortAnchorEl, setSortAnchorEl] = React.useState<null | HTMLElement>(null);
+  const [optionsAnchorEl, setOptionsAnchorEl] = React.useState<null | HTMLElement>(null);
 
-  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
-    setAnchorEl(event.currentTarget);
-  };
+  const dispatch = useDispatch();
+  const taskSettings = useSelector((state: StoreState) => state.settings.taskSettings);
+  const { sortBy } = taskSettings;
 
-  const handleClose = () => {
-    setAnchorEl(null);
+  const handleSortChange = (sort: string) => {
+    dispatch(setSettings({ taskSettings: { ...taskSettings, sortBy: sort } }));
+    setSortAnchorEl(null);
   };
 
+  const SortMenuItem = React.forwardRef(function SortMenuItem(props: { name: string, label: string }, _ref) {
```

Comment:
I'm a bit confused, why this ref forwarding? It doesn't look needed, and I don't know why you added it.

Reply:
There was an error in the console that I had missed: `Warning: Function components cannot be given refs`. I guess the `Menu` component sets a ref on its `MenuItem` children (see [MUI source code](https://github.com/mui-org/material-ui/blob/1159c65e31a6b992b7eda42fb63ccaa9aee5a946/packages/material-ui/src/Menu/Menu.js#L118-L127))

Reply:
I assume then you should forward the ref to the `MenuItem` and not just ignore it.

Reply:
yes, haha, just realized that

Reply:
fixed

Reply:
Grr, I just realised we have this warning in the side menu too (unrelated to your changes). I'll have to take a look later. :|